### PR TITLE
docs: Extend development tricks with dynamic test override

### PR DESCRIPTION
**Pull request by okurz**
Related progress issue: https://progress.opensuse.org/issues/36994

### CODE REVIEW

File: docs/WritingTests.asciidoc [hunk 1]
```diff
@@ -1192,3 +1195,48 @@ the clone source job or specifying the variable explicitly:
...
+Also test modules can be defined and overridden on-the-fly using a
+downloadable asset.
+
+For example one can schedule a job on a production instance with a custom
+schedule consisting of two modules from the provided test distribution plus
+one test module which is defined dynamically and downloaded as an asset from
+an external trusted download domain:
+
+----
+openqa-clone-job --from https://openqa.opensuse.org --host https://openqa.opensuse.org 24 SCHEDULE=tests/boot/boot_to_desktop,tests/console/consoletest_setup,foo ASSET_1_URL=https://download.opensuse.org/my/test/foo.pm
```
Comment (coolo) [inline docs/WritingTests.asciidoc]: however users get things on download.opensuse.org/my

Comment (okurz) [reply docs/WritingTests.asciidoc]: yes, it's just an example. I could give http://example.org if you prefer

Comment (coolo) [reply docs/WritingTests.asciidoc]: it would be more obvious IMO

File: docs/WritingTests.asciidoc [hunk 2]
```diff
@@ -1192,3 +1195,48 @@ the clone source job or specifying the variable explicitly:
 ----
 clone_job.pl --from https://openqa.opensuse.org  --host localhost 24 TESTDEBUG=1 SKIPTO=consoletest-yast2_i
 ----
+
+=== Defining a custom test schedule or custom test modules
+
+Normally the test schedule, that is which test modules should be executed and
+which order is prescribed by the +main.pm+ file within the test distribution.
```
Comment (Martchus) [inline docs/WritingTests.asciidoc]: I think inserting a comma after order would make it more readable (not sure about actual grammar rules in English).

File: docs/WritingTests.asciidoc [hunk 3]
```diff
@@ -1192,3 +1195,48 @@ the clone source job or specifying the variable explicitly:
...
+
+sub post_run_hook {}
+1;
+----
+
+This could be used e.g. in bug investigations, trying out new test modules
+which are hard to test locally, etc. I assume documentation is hard to find for
+people but I guess this is always the case.
+https://github.com/os-autoinst/os-autoinst/blob/master/doc/backend_vars.asciidoc
+describes the +SCHEDULE+ parameter as well as the others, the section "Asset
```
Comment (Martchus) [inline docs/WritingTests.asciidoc]: I would start a new sentence after the comma.

File: docs/WritingTests.asciidoc [hunk 4]
```diff
@@ -1192,3 +1195,48 @@ the clone source job or specifying the variable explicitly:
...
+handling" in the <<UsersGuide.asciidoc#usersguide,Users Guide>> describes how
+downloadable assets can be specified.  It is important to note that the
```
Comment (Martchus) [inline docs/WritingTests.asciidoc]: double white-space

File: docs/WritingTests.asciidoc (hunk 1 above)
Comment (okurz) [reply docs/WritingTests.asciidoc]: done

**Comment by okurz** [comment]
checks have passed now, ready to be merged?

File: docs/WritingTests.asciidoc [hunk 5]
```diff
@@ -1248,3 +1251,48 @@ the clone source job or specifying the variable explicitly:
...
+    select_console 'root-console';
+    assert_script_run 'foo';
+}
+
+sub post_run_hook {}
+1;
+----
+
+This could be used e.g. in bug investigations, trying out new test modules
+which are hard to test locally, etc. I assume documentation is hard to find for
```
Comment (foursixnine) [inline docs/WritingTests.asciidoc]: I've read it 3 times, still reads a bit weird, can you improve the wording? I'll merge after, unless somebody else merges ;)

**Comment by foursixnine** [comment]
Also #1853 is related :) (somehow ;))

File: docs/WritingTests.asciidoc (hunk 5 above)
Comment (okurz) [reply docs/WritingTests.asciidoc]: rephrased, please check again

**Comment by okurz** [comment]
Yes, https://github.com/os-autoinst/openQA/pull/1853 is related but not required

**Pull request merged**